ci/test-stable.sh

```diff
@@ -12,6 +12,30 @@ annotate() {
   }
 }
 
+cargo_build_sbf_sanity() {
+  cargo_build_sbf="$(realpath ./cargo-build-sbf)"
+  # These programs must match those available on the list:
+  # https://github.com/anza-xyz/agave/blob/e12e6fbf76a0a3ba6acc85c3ef43467a7f90f38b/programs/sbf/tests/programs.rs#L149-L169
+  local sanity_programs=("128bit" "alloc" "alt_bn128" "alt_bn128_compression" "curve25519" "custom_heap" "dep_crate"
```

Comment:
Can't this list be gathered automatically?
Otherwise it is bound to rot over time.

Reply:
> Otherwise it is bound to rot over time.

If the sanity programs change the CI will fail, since we call the test `test_program_sbf_sanity` specifically. If the program isn't in the bash script, the test won't find the binary. On the other hand, if someone removes a program from the test, we would be building more programs than necessary and potentially missing edge cases.

> Can't this list be gathered automatically?

I believe better than this would be building every rust program with `cargo-build-sbf`, so that we ensure the tool is working correctly. I didn't take this approach since you removed it because it was slow, wasn't it?

Reply:
Since all programs are in the same workspace, running `cargo-build-sbf` separately for each one of them will still reuse most of the intermediate objects, so time shouldn't be an issue.

Reply:
`stable-sbf` took 24 minutes in this PR. Let me re-work it and check the time again.

Reply:
Building and running everything took 26 minutes.